src/providers/wms/qgsxyzconnection.cpp

```diff
@@ -41,8 +42,27 @@ QString QgsXyzConnection::encodedUri() const
 QStringList QgsXyzConnectionUtils::connectionList()
 {
   QgsSettings settings;
+  QStringList connList;
+
   settings.beginGroup( QStringLiteral( "qgis/connections-xyz" ) );
-  return settings.childGroups();
+  connList = settings.childGroups();
+
+  QStringList global = settings.globalChildGroups();
+  settings.endGroup();
+
+  for ( auto &s : global )
```

Comment:
To avoid a detach you might want to try the new `qgsAsConst` https://github.com/qgis/QGIS/commit/029f741cfaac27b9e1959629afe454faed46e0ca recently added by @nyalldawson or use `Q_FOREACH`

Reply:
It also should be `const auto &`

Reply:
> To avoid a detach you might want to try the new qgsAsConst 029f741 recently added by @nyalldawson or use Q_FOREACH

... or simply use `const QStringList global`

Reply:
I prefer to [avoid the `Q_FOREACH`](https://www.kdab.com/goodbye-q_foreach/).

I've used the suggested pattern:
```CPP
  const QStringList global = settings.globalChildGroups();
  for ( const auto &s : global )
```